src/main/java/com/vinylteam/vinyl/web/servlets/EditProfileServlet.java

```diff
@@ -66,7 +66,12 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
                 } else {
                     boolean checkOldPassword = securityService.checkPasswordAgainstUserPassword(user, oldPassword.toCharArray());
                     if (checkOldPassword) {
-                        boolean isEdit = userService.update(email, newEmail, newPassword, newDiscogsUserName);
+                        boolean isEdit;
+                        if (!newPassword.equals("")){
+                            isEdit = userService.update(email, newEmail, newPassword, newDiscogsUserName);
+                        } else {
+                            isEdit = userService.update(email, newEmail, oldPassword, newDiscogsUserName);
+                        }
                         logger.debug("Got result of edit user by email with " +
```

Comment:
It's better to put logger in if - else:
logger.info("User was updated with new email or password in the database {'newEmail':{}}.", newEmail);

and

logger.info("Failed to update with new email or password user in the database {'newEmail':{}}.", newEmail);